app/assets/javascripts/discourse/app/components/post/menu/button-wrapper.gjs

```diff
@@ -0,0 +1,19 @@
+import Component from "@glimmer/component";
+
+// eslint-disable-next-line ember/no-empty-glimmer-component-classes
+export default class PostMenuButtonWrapper extends Component {
+  // we need a class component because we need to pass this.args to the config helpers
+
+  <template>
+    <@buttonConfig.Component
+      class="btn-flat"
+      ...attributes
+      @alwaysShow={{@buttonConfig.alwaysShow this.args}}
+      @buttonActions={{@buttonActions}}
+      @context={{@context}}
+      @post={{@post}}
+      @shouldRender={{@buttonConfig.shouldRender this.args}}
```

Comment:
The shouldRender architecture feels a little weird to me. Buttons define a static `shouldRender` method, which is evaluated outside the component, then passed back into the button component, which then has to wrap itself in a `{{#if shouldRender}}`

IMO it would be much cleaner if the `{{#if shouldRender}}` check was done in this ButtonWrapper component, so that the button component is not rendered when shouldRender is false.

IIRC the current architecture is so that themes/plugins can replace the component, and choose to ignore the `shouldRender` value? Maybe we could make a way for them to override the shouldRender function instead?

Reply:
Also, rendering every single button component even when it's not required is quite a large cost. If we can apply the shouldRender logic earlier, it should make things faster.

Reply:
Your point about the `shouldRender` logic being too convoluted and the performance issues, makes total sense.

I'll think of a way to prevent the buttons from being rendered, and if the user chooses to override the shouldRender static method, we still be able to provide the core `shouldRender` value to the component.

That would allow it to disabled it instead of hiding for example

Reply:
I introduced a new static property in the component called `delegateShouldRenderToTemplate`. When it's set to `true` the component is rendered and the value of `shouldRender` is passed as an argument.

This solves both issues, we no longer render the buttons and check if it should be rendered in the template, and as consequence, we can remove the  `{{#if @shouldRender}}` from the core buttons